### PR TITLE
Add the local PV path provisioner

### PR DESCRIPTION
**Pull request by meyskens**

### CODE REVIEW

**Comment by meyskens** [comment]
Still needs long running tests (watchers have some weird behavior after a while)

File: cmd/k8s-local-pv-provisioner/run.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+)
+
+func init() {
+	app.AddCommand(&RunCommand{})
+}
+
+type RunCommand struct {
+	gocli.PlainCommand `name:"run" short-description:"run a watcher for PVs" long-description:"Run an in-cluster watcher for PVs and create the needed paths if needed"`
+	NodeName           string `long:"node-name" required:"true" env:"NODE_NAME" description:"Hostname of the current node the pod runs on"`
+	KubernetesContext  string `long:"context" env:"KUBERNETES_CONTEXT" description:"If set the program will oad the kubernetes configuraion from a kubeconfig file for the given context"`
```
Comment (rporres) [inline cmd/k8s-local-pv-provisioner/run.go]:
```suggestion
	KubernetesContext  string `long:"context" env:"KUBERNETES_CONTEXT" description:"If set the program will load the kubernetes configuration from a kubeconfig file for the given context"`
```

File: cmd/k8s-local-pv-provisioner/run.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+func (r *RunCommand) setUpPV(pv *core_v1.PersistentVolume) {
+	if pv.Spec.Local == nil {
+		log.Infof("%s is not a local volume, skipping", pv.ObjectMeta.Name)
+		return
+	}
+
+	matches := false
+	// We can assume this is not nil as `Spec.Local` is invalid without NodeAffinity
```
Comment (rporres) [inline cmd/k8s-local-pv-provisioner/run.go]: But somebody could force creating a resource without validation and that would make the whole program crash, isn't it?

File: cmd/k8s-local-pv-provisioner/run.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	for _, term := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms {
+		for _, matchExpression := range term.MatchExpressions {
+			if matchExpression.Key != "kubernetes.io/hostname" { // we only have hostname support
+				continue
+			}
+			if matchExpression.Operator == core_v1.NodeSelectorOperator("In") {
+				for _, value := range matchExpression.Values {
+					if value == r.NodeName { // replace me with an env var!!!!
```
Comment (rporres) [inline cmd/k8s-local-pv-provisioner/run.go]: I don't fully understand this comment. How would a variable avoid the fact of checking that this executable should only care about those PVs related to its own NodeName?

File: manifests/daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: k8s-local-pv-provisioner
+        name: k8s-local-pv-provisioner
+    spec:
+      serviceAccountName: k8s-local-pv-provisioner
+      hostNetwork: true
```
Comment (rporres) [inline manifests/daemonset.yaml]: why do we need this to be true?

**Review by rporres — COMMENTED** [review]
Looks great. A few comments here and there

File: cmd/k8s-local-pv-provisioner/run.go (hunk 3 above)
Comment (rporres) [reply cmd/k8s-local-pv-provisioner/run.go]: Or is it related to the fact that you can only have a reliable node name in host network?

Comment (rporres) [reply cmd/k8s-local-pv-provisioner/run.go]: Ok, taking a look to https://github.com/src-d/charts/pull/130/files it definitely looks the last option, so please go that route

**Comment by rporres** [comment]
Why do we want to keep k8s templates here if we have a chart?

**Comment by rporres** [comment]
A README would also be great 😅

File: cmd/k8s-local-pv-provisioner/run.go (hunk 2 above)
Comment (meyskens) [reply cmd/k8s-local-pv-provisioner/run.go]: True, will build in a check for that! 
I forgot you can turn validation off.

File: cmd/k8s-local-pv-provisioner/run.go (hunk 3 above)
Comment (meyskens) [reply cmd/k8s-local-pv-provisioner/run.go]: A lost development comment :disappointed: I did replace it but forgot to remove the note to myself

File: manifests/daemonset.yaml (hunk 4 above)
Comment (meyskens) [reply manifests/daemonset.yaml]: Can be deleted, overlooked this

**Comment by meyskens** [comment]
I think the templates can be an extra for the open source part, especially as not everyone uses helm.
I will also add a readme soon :smile:

File: cmd/k8s-local-pv-provisioner/run.go (hunk 3 above)
Comment (rporres) [reply cmd/k8s-local-pv-provisioner/run.go]: Cool. `vim` highlights the `TODO` in a nasty yellow so that I never forget :-D

**Comment by rporres** [comment]
>  I think the templates can be an extra for the open source part, especially as not everyone uses helm.

As long as both things are in-sync, I'm totally fine :)

File: cmd/k8s-local-pv-provisioner/run.go (hunk 3 above)
Comment (meyskens) [reply cmd/k8s-local-pv-provisioner/run.go]: Hmm i heard that before, need to check if I can get that in VSCode
(a terrible slippy style lookout analyzer idea came to mind... but let's not go there)

**Comment by meyskens** [comment]
Also we need to run dep :cold_sweat:

**Review by rporres — APPROVED** [review]

**Pull request merged**